fastlane/lanes/release.rb

```diff
@@ -118,31 +221,75 @@
+    hotfix_build_code = hotfix_build_code(release_version: new_version)
+
+    # Parse the provided version into an AppVersion object
+    parsed_version = VERSION_FORMATTER.parse(new_version)
+    previous_version = VERSION_FORMATTER.release_version(VERSION_CALCULATOR.previous_patch_version(version: parsed_version))
+
+    # Check versions
+    message = <<-MESSAGE
+
+      New Hotfix:
+
+      • Current release version and build code: #{current_release_version} (#{current_build_code}).
+      • New release version and build code: #{new_version} (#{hotfix_build_code}).
+
+      • Current internal release version and build code: #{current_internal_release_version} (#{current_internal_build_code}).
+      • New internal release version and build code: #{new_version} (#{internal_hotfix_build_code(release_version: new_version)}).
+
+      Branching from tag: #{previous_version}
+
+    MESSAGE
+
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
+
+    # Check tags
+    UI.user_error!("Version #{new_version} already exists! Abort!") if git_tag_exists(tag: new_version)
+    UI.user_error!("Version #{previous_version} is not tagged! A hotfix branch cannot be created.") unless git_tag_exists(tag: previous_version)
+
+    # Create the hotfix branch
+    UI.message 'Creating hotfix branch...'
+    ensure_git_status_clean
+    Fastlane::Helper::GitHelper.create_branch("release/#{new_version}", from: previous_version)
+    UI.message "Done! New hotfix branch is: #{git_branch}"
+
+    # Bump the hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping hotfix version and build code...'
+    PUBLIC_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: hotfix_build_code
+    )
+    UI.message "Done! New Release Version: #{current_release_version}. New Build Code: #{current_build_code}"
+
+    # Bump the internal hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping internal hotfix version and build code...'
+    INTERNAL_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: internal_hotfix_build_code
     )
+    UI.message "Done! New Internal Release Version: #{current_internal_release_version}. New Internal Build Code: #{current_internal_build_code}"
+
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
   end
 
   # Finalizes a hotfix, by triggering a release build on CI
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
   #
   desc 'Performs the final checks and triggers a release build for the hotfix in the current branch'
-  lane :finalize_hotfix_release do |options|
-    ios_finalize_prechecks(options)
+  lane :finalize_hotfix_release do
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
```

Comment:
To me, the code is already self explanatory:

```suggestion
    ensure_git_branch(branch: '^release/')
```

Reply:
As a side note, the main reason why we have a similar comment in Tumblr Fastfiles is because up until a month ago or so, we used `ensure_git_branch(branch: 'develop')` for example as a way to check if we were on the `develop` branch at the start of a lane… and then later were baffled to find out that a lane created a PR that included more than it should as if it had been created from the wrong branch.

Turned out, the wrong branch the PR was created from was named `merge/beta-xxx-into-develop`… which happens to be a branch name that passed the `ensure_git_branch(branch: 'develop')` test. That's only then that we dug into that action's implementation in _fastlane_ and realized that it was not checking for branch name equality via `==`, but instead the parameter was supposed to be a regex pattern, and checked the branch using `.match`. So we updated all our calls from `ensure_git_branch('develop')` to `ensure_git_branch('^develop$')` instead to not be bitten by this subtlety anymore.

I feel like it's quite obscure and not well known (nor documented) that `ensure_git_branch` expects a RegEx, and I believe that's where the `# Match branch names that begin with 'release/'` comment comes from. Not to explain what the `^release/` RegExp does, but to highlight that this parameter is expected to be a RegExp in the first place, and that using something like `release/*` like one might think (and like what we used to before) would be incorrect.

Reply:
@mokagio I agree with @AliSoftware here that it may be helpful to have this comment here for clarity about is being passed to `ensure_git_branch`. I could also update the comment to something like:

A RegEx is passed to `ensure_git_branch` to verify that the branch begins with `release/`

Reply:
@AliSoftware gotcha, thank for the context. Also, TIL that the parameter is _expected_ to be a RegEx.

@spencertransier 

> I could also update the comment to something like:
>
> A RegEx is passed to ensure_git_branch to verify that the branch begins with release/

Makes sense. What do you think of 

```suggestion
    # Notice that ensure_git_branch expects a RegEx parameter. 
    ensure_git_branch(branch: '^release/')
```

Reply:
Another option, which may or may not be taking it too far, would be to define our self-documented wrapper:

```ruby
ensure_git_branch(regex: /^release\//)
```

Reply:
My preference would be for us to make a PR in _fastlane_ itself to fix it at the root though.

The tricky aspect is we'd have to make sure our change to that action in _fastlane_ is backwards compatible, so we can't just change the existing `ConfigItem(key: :branch)` to behave with `==` when passed a string (while behaving with `Regexp#match` when passed a `Regexp` or when using `regex:` instead of `branch:` param) because that would be changing behavior and breaking compatibility for users of this action using `(branch: 'some regex')`.

So instead I'm thinking we could change the action to add a parameter like `ConfigItem(key: :use_regexp_matching, type: Boolean, default: true)`, and have the implementation use `==` if false, while use `Regexp#match` like today if true; which would ensure we keep backwards compatibility, all while being able to use `ensure_git_branch(branch: '^release/', use_regex_matching: true)` in our code (even if true would always be the default) as a way to be self-documenting.

---

Another alternative would be to modify the existing action to add both `ConfigItem(key: :exact, type: String, conflicting_options: [:branch, :regexp])` and `ConfigItem(key: :regexp, type: Regexp, conflicting_options: [:branch, :exact])`, then mark the `:branch` parameter as deprecated.

Reply:
For now, I updated the comment for these checks to: 

`Verify that the current branch is a release branch. Notice that `ensure_git_branch` expects a RegEx parameter`

Commit: https://github.com/wordpress-mobile/WordPress-iOS/commit/418c28c752c856327a36f3bf7e34083fdf73f2fd